.github/workflows/docker-build-push.yaml

```diff
@@ -7,9 +7,25 @@ on:
     tags:
       - "v*.*.*"
   pull_request:
+  workflow_dispatch:
+  issue_comment:
+    types: [created]
 jobs:
   docker-pipeline:
     runs-on: ubuntu-latest
+    if: |
+        (github.event_name == 'push' && (
+          github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v')
+          )
+        ) ||
+        github.event_name == 'pull_request' ||
+        github.event_name == 'workflow_dispatch' ||
+        (github.event_name == 'issue_comment' &&
+        github.event.issue.pull_request &&
+        contains(github.event.comment.body, '/test') &&
+        (github.event.comment.author_association == 'MEMBER' ||
```

Comment:
Are we attempting to implement AuthZ logic in here? Is this tackled already within GitHub?

Reply:
PS: Does this solve our needs? https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

Reply:
Let me give this workflow run a try. I'd prefer that if it works since the 'managed' workflow would actually show up in the PR UI (and failing would block) unlike how it works on a comment based trigger